Sources/Workspace/LocalContainerProvider.swift

```diff
@@ -0,0 +1,111 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import PackageModel
+import PackageGraph
+import TSCBasic
+import TSCUtility
+import SourceControl
+
+/// PackageContainerProvider implementation used by Workspace to do a dependency pre-calculation using the cached
+/// dependency information (Workspace.DependencyManifests) to check if dependency resolution is required before
+/// performing a full resolution.
+struct LocalContainerProvider: PackageContainerProvider {
+    let root: PackageGraphRoot
+    let currentManifests: Workspace.DependencyManifests
+    let config: SwiftPMConfig
+    let diagnostics: DiagnosticsEngine
+
+    func getContainer(
+        for identifier: PackageReference,
+        skipUpdate: Bool,
+        completion: @escaping (Result<PackageContainer, AnyError>) -> Void
+    ) {
+        if let manifest = currentManifests.dependencies.first(where: { $1.packageRef == identifier }) {
```

Comment:
Maybe add comment explaining what you're doing here?

Reply:
Done.